### ISSUE TITLE
mapPlot(coastlineWorld) fails in Windows build - request of other developers

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1492]
I decided I ought to build up windows and OSX binaries of today's oce, so that folks can get access to labelled `mapContour()` without building from source. In doing that, I also removed the `\dontrun{}` blocks in the example sections of the docs for `mapContour()` and `magneticField()`. 

Unfortunately, removing those `\dontrun{}` blocks yields build failures when I run `devtools::check_win_release()`. The code
```R
library(oce)
data(coastlineWorld)
mapPlot(coastlineCut(coastlineWorld, lon_0=100), proj="+proj=moll +lon_0=100", col='gray')
```
gives the error message
```
Error in rgdal::project(ll, proj = as.character(projection), inv = FALSE) : 
  failure in points 
Calls: mapPlot ... withVisible -> eval -> eval -> unname -> <Anonymous>
Execution halted
```
Since I do not have direct access to a windows machine, I really have no idea how to proceed in trying to find out the problem. At first, I thought it was a problem only with the `projection="+proj=robin"` that had been my first choice, but it fails also with `+proj=moll`, as shown above.

I wonder if another developer would be willing to try some tests on this? A test with the world coastline is
```R
xy <- rgdal::project(cbind(coastlineWorld[["longitude"]],coastlineWorld[["latitude"]]), proj="+proj=moll", inv=FALSE)
```
and a test with a single point is
```R
rgdal::project(cbind(0,0), proj="+proj=moll", inv=FALSE)
```
Unfortunately, I don't know whether either of these will cause the error to occur. (It could be something about transforming lines of constant lon or lat, for example.)  But, hopefully, it will be possible to narrow down on the error by running the original code interactively, and then the other two tests.  (For example, running `traceback()` ought to yield line numbers in `R/map.R`, if the source was built locally.)

It would be a shame if windows users couldn't draw a map with projection.

**Comment by richardsc** [comment]
I'll take a look when I can. Maybe tonight or tomorrow sometime.

**Comment by dankelley** [comment]
I decided to look up the error message ("failure in points"), by doing
```
git svn clone svn://scm.r-forge.r-project.org/svnroot/rgdal/
git grep -n "failure in points" pkg
```
then noting the results from the latter, viz
```
pkg/R/project.R:54:       stop(paste("failure in points", paste(k, collapse=":")))
pkg/R/project.R:69:       stop(paste("failure in points", paste(k, collapse=":")))
pkg/R/project.R:163:            stop(paste("failure in points", paste(k, collapse=":")))
pkg/R/project.R:176:            stop(paste("failure in points", paste(k, collapse=":")))
```
I had a look in `pkg/R/project.R` and saw as in the screenshot.  This is a bit disconcerting. Notice that the error message I got (at https://win-builder.r-project.org/yfsj17XG9H72/00check.log but that link will disappear in a day or so) has *no* index numbers stated, although the error messages all are supposed to list the bad `k` values. But look at the `which()` call in line 51, which is wrongly using `||` instead of the correct `|`.  I'll look into how to report that problem (although, of course, it is only in how the error message is reported; I don't think it relates to the cause of our current `oce` problem).

<img width="655" alt="screen shot 2019-02-18 at 10 18 56 am" src="https://user-images.githubusercontent.com/99469/52956838-a934c780-3366-11e9-8181-53b44a531952.png">
